### DISCUSSION TITLE
TI SimpleLink CC13xx and CC26xx SDK 5.30 / 5.30.00.56 released with support for newer CC13xx and CC26xx hardware chips

### DISCUSSION THREAD

**Discussion by Hedda** [Koenkk/zigbee2mqtt#9744] [General]
FYI, apparently, Texas Instruments has released SimpleLink CC13xx and CC26xx SDK v5.30 / version 5.30.00.56 on the 27th of October 2021:

https://www.ti.com/tool/SIMPLELINK-CC13XX-CC26XX-SDK

This newest release with a new SDK name replaces SimpleLink CC13x2 / CC26x2 SDK versions 5.20 and earlier!

Many probably missed that v5.30 had already been released because going from v5.20 to v5.30 TI has renamed their Software Development Kit (SDK) from "SimpleLink CC13x2 and CC26x2 SDK" to "SimpleLink CC13xx and CC26xx SDK" as it is now also usable for newer devices.

Some new devices include CC2652P7, CC2652R7, CC1352P7, CC2651P3, and CC2651R3 (with **704 kB Flash memory** and 152 kB SRAM + cache memory), plus new CC2652RSIP and CC2652PSIP platforms  (with 352 kB Flash memory and 88 kB SRAM + cache memory):

https://www.ti.com/product/CC2652RSIP (7-mm-by-7-mm system-in-package wireless module)

https://www.ti.com/product/CC2652PSIP (System-in-package module with integrated power amplifier)

https://ti.com/product/CC2652P7 (**Higher memory**, multiprotocol wireless MCU with an integrated power amplifier)

https://ti.com/product/CC2652R7 (**Higher memory**, multiprotocol wireless MCU)

https://ti.com/product/CC1352P7 (Multiprotocol and multiband wireless MCU with an integrated power amplifier)

TI SimpleLink CC13xx and CC26xx SDK 5.30 / 5.30.00.56 release notes:

https://www.ti.com/tool/download/SIMPLELINK-CC13XX-CC26XX-SDK#release-notes

## _What's New?_
* _This SDK has been renamed to SIMPLELINK-CC13XX-26XX-SDK to reflect new devices supported by the SDK. This SDK is backwards compatible for qualification purposes with previous SDK versions. CC13x0 and CC2640R2 devices will continue to use their respective SDKs (CC13X0 SDK, CC2640R2 SDK)._
* _Proprietary RF examples have moved from the drivers/ folder to a new proprf/ folder._
* _This is the initial SDK release with TI-RTOS7 examples. See note below for more information._
* _New FreeRTOS examples for additional stacks._
* _New GPIO++ driver._
* _The TI Wi-SUN FAN Border Router has been certified with the Wi-SUN Alliance._
* _New SimpleLink Common Component Modules (SCCM) release notes._

_Refer to the Document Overview for the individual release notes and details for starting development with each SDK component._

_Refer to the included Change Log for a summary of new features and fixed issues since the last SDK release._

https://software-dl.ti.com/simplelink/esd/simplelink_cc13xx_cc26xx_sdk/5.30.00.56/exports/changelog.html

https://software-dl.ti.com/simplelink/esd/simplelink_cc13xx_cc26xx_sdk/5.30.00.56/exports/docs/Documentation_Overview.html

### _Supported products & hardware:_

_cc13xx_cc26xx
CC1311P3
CC1311R3
CC1312R
CC1312R7
CC1352P
CC1352P7
CC1352R
CC2642R
CC2642R-Q1
CC2651P3
CC2651R3
CC2652P
CC2652P7
CC2652R
CC2652RB
CC2652RSIP_

#### _HARDWARE_

_LAUNCHXL-CC1312R1 — SimpleLink™ Sub-1 GHz CC1312R Wireless Microcontroller (MCU) LaunchPad™ Development Kit
LAUNCHXL-CC1352P — SimpleLink™ multi-band CC1352P wireless MCU LaunchPad™ development kit
LAUNCHXL-CC1352R1 — SimpleLink™ Multi-Band CC1352R Wireless MCU LaunchPad™ Development Kit
LAUNCHXL-CC26X2R1 — SimpleLink™ multi-standard CC26x2R wireless MCU LaunchPad™ development kit
LP-CC2652RB — SimpleLink™ crystal-less BAW CC2652RB multiprotocol 2.4GHz wireless MCU LaunchPad™ development kit
LP-CC2652RSIP — SimpleLink™ multi-protocol CC2652RSIP wireless module LaunchPad™ development kit
LPSTK-CC1352R — SimpleLink™ multi band CC1352R wireless MCU Launchpad™ SensorTag kit_

**Comment by Hedda** [comment]
My guess is that new CC2652P7 and CC2652R7 chips with higher memory are designed to run more than one protocol at the same time?

Do you think that these could support running Z-Stack ZNP and Thread / OpenThread on the same chip and run both at the same time? 

That is, have the same dongle with only one CC2652P7 or CC2652R7 chip act as both a Zigbee Coordinator and a Thread Border Router?

**Comment by Koenkk** [comment]
FYI; firmwares with this SDK are already available: https://github.com/Koenkk/Z-Stack-firmware/tree/develop/coordinator/Z-Stack_3.x.0

**Reply by Hedda** [reply]
Ah, nice! Any news about a firmware with +20 dBm output power support on CC2652P and CC1352P based radio modules? 

As per this request -> https://github.com/Koenkk/Z-Stack-firmware/issues/323

Personally, I'm requesting 20 dBm transmit power working on the new [Sonoff Zigbee 3.0 USB Dongle Plus adapter by ITead](https://github.com/Koenkk/zigbee2mqtt/discussions/8840).

**Reply by Koenkk** [reply]
I have had contact with Sonoff about this, they are currently checking whether it is possible to set a default transmit power of 9dbm (to pass certifications), waiting for them to reply

**Reply by Hedda** [reply]
Does that dBm transmit power tuning depends on the antenna they use when testing for certifications? ETSI compliance EN 300 328 right?

I did not know that ITead was going for certifications, I also believed the general limit was +10 dBm in RF TX power for 2.4GHz "wide band modulation" Western Europe for CE certification and that it included WiFi, Bluetooth, IEEE 802.15.4/ZigBee, etc.?

Does anyone know if maximum dBm transmission power specifically for 2.4GHz is different for FCC (North America) and CE (Western Europe)?

https://www.digi.com/resources/documentation/Digidocs/90000991/reference/r_certs_xb_europe.htm

"**_The power output of the XBee-PRO RF Modules must not exceed 10 dBm. The power level is set using the PL command. The International Variant of this product is internally limited to 10 dBm._**"

**Reply by Koenkk** [reply]
This is what I got from them:

> 1. As the certifier recommends us to set the default transmit power as 9.61 dbm, which need to lower than 10. You may take this as a reference to adjust your plan?
